package.json

```diff
@@ -1,25 +1,32 @@
 {
   "name": "@novnc/novnc",
-  "version": "1.6.0",
+  "version": "1.6.1-alpha.1",
   "description": "An HTML5 VNC client",
-  "browser": "lib/rfb",
-  "directories": {
-    "lib": "lib",
-    "doc": "docs",
-    "test": "tests"
-  },
+  "type": "module",
   "files": [
-    "lib",
+    "core",
+    "vendor",
     "AUTHORS",
     "VERSION",
     "docs/API.md",
     "docs/LIBRARY.md",
     "docs/LICENSE*"
   ],
+  "exports": {
+    ".": "./core/rfb.js",
+    "./lib/*": "./core/*.js",
+    "./lib/*.js": "./core/*.js",
+    "./core/*": "./core/*.js",
+    "./core/*.js": "./core/*.js",
+    "./lib/vendor/*": "./vendor/*.js",
+    "./lib/vendor/*.js": "./vendor/*.js",
+    "./vendor/*": "./vendor/*.js",
+    "./vendor/*.js": "./vendor/*.js",
+    "./package.json": "./package.json"
+  },
```

Comment:
RFB is our only interface. Do we need anything more than the first line?

Reply:
There are uses I could find online using the github search:

with lib: https://github.com/SDU-eScience/UCloud/blob/e78be592106cd915bb9004fc095daa5140e23ed1/frontend-web/webclient/app/Applications/Jobs/Vnc.tsx#L13

https://github.com/kubevirt-ui/kubevirt-plugin/blob/447047ed328719698db1f9e1de5f2db81dc40b89/src/utils/components/Consoles/components/vnc-console/VncConsole.tsx#L6

with core: https://github.com/harvester/dashboard/blob/6da3d8db703521eef18cb30ad60c4148bba4edff/pkg/harvester/components/novnc/NovncConsoleWrapper.vue#L5

And since I do not want these inputs to break I have written it as it is there.

I could probably remove the part about vendor, since it does not seem to be used.

Reply:
Those are all unsupported usages of noVNC. I'm not keen on giving any illusion of that being official and stable.

Ideally, we just have:

```js
  "exports": "./core/rfb.js",
```

Which is the only supported interface at the moment.

The question is what that transition means for people. Given that we are switching from CommonJS to modules, I assume there is some transition work needed anyway? So might be a good idea to do this at the same time?

Reply:
`util/logging.js` is clearly also exposed api, otherwise, the provided example vnc.html would not work:

https://github.com/novnc/noVNC/blob/4cb5aa45ae559f8fa85fe2b424abbc6ef6d4c6f9/vnc.html#L53

Reply:
As are the files under `input/key*`:
https://github.com/novnc/noVNC/blob/4cb5aa45ae559f8fa85fe2b424abbc6ef6d4c6f9/app/ui.js#L15-L17

Reply:
Exposed is not the same as supported or stable. I.e. you use them at your own risk. We will not consider it a bug if we change them in a way that breaks callers.

For our internal use, that is fine as we can change it at the same time. But external users cannot really do the same safely.

Reply:
I'm aware this will cause some pain for some users. But that might be the best thing in the long run, as that will force a discussion of what is missing from the official API and needs to be added.